guide/src/class/object.md

```diff
@@ -70,6 +70,46 @@ impl Number {
 }
 ```
 
+To automatically generate the `__str__` implementation using a `Display` trait implementation, pass the `str` argument to `pyclass`.
+
+```rust
+# use std::fmt::{Display, Formatter};
+# use pyo3::prelude::*;
+#
+# #[pyclass(str)]
+# struct Coordinate {
+    x: i32,
+    y: i32,
+    z: i32,
+}
+
+impl Display for Coordinate {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "({}, {}, {})", self.x, self.y, self.z)
+    }
+}
+```
+
+For convenience, a shorthand format string can be pass to `str` as `str="<format string>"`.  It expands and is passed into the `format!` macro in the following ways:  
+
+* `"{x}"` -> `"{}", self.x`
+* `"{0}"` -> `"{}", self.0`
+* `"{x:?}"` -> `"{:?}", self.x`
+* `"{:?}"` -> `"{:?}", self`
```

Comment:
Hmm, I'm not sure how I feel about this one. Not specifying a ident here feels unintuitive to me. I'm also not sure how useful access to `self` here is...

Reply:
The main benefit for access to self is on enums.  It would be how you format the variant.  For instance, in the current simple enum case, we implement repr to be "<cls>.<variant_name>".  Rust's default for debug would be just to name the variant, not the cls in the output.  The current simple enum behavior can be retained by doing something like this:

```rust
#[pyclass(eq, str = "MyEnum.{:?}")]
#[derive(Debug, PartialEq)]
pub enum MyEnum {
    Variant,
    OtherVariant,
}
```
I am not sure how else we might declare this with a shorthand format without the user needing to implement Display directly.  Though maybe we should force them to that for simple enums (so removing the shorthand format string entirely for simple enums).  They only downside I see to that, is if they want to have a different formatting applied on the Python side vs the Rust side (though they could implement both __str__ and Display explicitly to achieve it).  I suppose we have to strike the right balance for flexibility vs. intuition while retaining decent ergonomics.